datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ResolverUtils.java

```diff
@@ -55,4 +65,47 @@ public static String escapeForwardSlash(@Nonnull String input) {
 
         return facetFilters;
     }
+
+    @Nonnull
+    public static AspectWithMetadata getAspectFromLocalContext(DataFetchingEnvironment environment) {
+        String fieldName = environment.getField().getName();
+        Long version = environment.getArgument("version");
+
+        Object localContext = environment.getLocalContext();
+        // if we have context & the version is 0, we should try to retrieve it from the fetched entity
```

Comment:
Interesting ... Do we need this is we are only using for schemaMetadata case, where we are sure we need a specific version?

Reply:
Not sure I understand this question- this resolver is for all versioned aspect fetchers. If someone fetches version 0, we don't actually need to use the aspect resource since version 0 has already been fetched earlier by the Entity type's batchGet.

Reply:
Hmmm. Does it make sense to have a versioned aspect fetcher everywhere? I guess that's my main question